pkg/arvo/app/radar.hoon

```diff
@@ -0,0 +1,304 @@
+/+  *server
+|%
++$  move  [bone card]
++$  card
+  $%  [%poke wire dock pear]
+      [%wait wire @da]
+      [%rest wire @da]
+      [%info wire toro:clay]
+      [%connect wire binding:eyre term]
+      [%http-response =http-event:http]
+  ==
+::
++$  pear
+  $%  [%noun cord]
+      [%helm-hi cord]
+  ==
+::
++$  state
+  $:  scanning=_|                        :: are we currently scanning?
+      filter=(set ship)
+      configured=(set ship)              :: set of ships to scan
+      outstanding=(map ship @da)         :: ships we've |hi'd with no reply
+      log=(map ship (list [@da @da]))    :: log of past |hi's sent/recv by ship
+      timers=(map wire @da)              :: map of delay timers
+      moves=(list move)                  :: list of outgoing moves
+  ==
+::
+--
+|_  [bol=bowl:gall sty=state]
+::
+++  this  .
+::
+++  abet
+  ^-  (quip move _this)
+  [moves.sty this(moves.sty ~)]
+::
+++  bound
+  |=  [wir=wire success=? binding=binding:eyre]
+  ^-  (quip move _this)
+  [~ this]
+::
+++  prep
+  |=  old=(unit state)
+  ^-  (quip move _this)
+  :-  [[ost.bol %connect / [~ /'~radar'] %radar]]~
+  ?~  old
+    %=    this
+    ::
+        scanning.sty  %.y
+    ::
+        configured.sty  ~
+    ==
+  this(sty u.old)
+::
+++  get-point-json
+  |=  her=ship
+  ^-  json
+  ?.  (~(has in configured.sty) her)
+   [%s 'not being tracked']
+  =/  lastseen/(unit @da)
+    (~(get by outstanding.sty) her)
+  =/  log/(list [@da @da])
+    (fall (~(get by log.sty) her) ~)
+  =/  failed/(list {p/@t q/json})
+    ?~  lastseen
+      ~
+    [%'last-seen' (time:enjs:format u.lastseen)]~
+  %-  pairs:enjs:format
+  :_  failed
+  :-  %responded
+  :-  %a
+  %+  turn  log
+  |=  [sent=@da recv=@da]
+  ^-  json
+  %-  pairs:enjs:format
+  :~  [%sent (time:enjs:format sent)]
+      [%recv (time:enjs:format recv)]
+  ==
+++  poke-noun
+  |=  a=*
+  ^-  (quip move _this)
+  ?+  a
+    [~ this]
+  ::
+      %running
+    ~&  scanning.sty
+    [~ this]
+  ::
+      %print
+    ~&  sty
+    [~ this]
+  ::
+      %start
+    ?:  scanning.sty
+      ~&  'already running'
+      [~ this]
+    =.  this  this(scanning.sty &)
+    ::
+    =.  this  (scan ~(tap in configured.sty))
+    abet
+  ::
+      %stop
+    :_  this(scanning.sty |, timers.sty ~)
+    %-  ~(rep by timers.sty)
+    |=  [[wir=wire d=@da] mov=(list move)]
+    ^-  (list move)
+    [[ost.bol %rest wir d] mov]
+  ::
+      %flush
+    [~ this(log.sty ~)]
+  ==
+::
+
+++  ping
+  |=  her=ship
+  ^+  this
+  %=  this
+  ::
+      moves.sty
+    :_  moves.sty
+    :*  ost.bol
+        %poke
+        /ping/(scot %p her)/(scot %da now.bol)
+        [her %hall]  %noun  'fdsjfk'
```

Comment:
%hall is being removed so we should poke another app. If we're intending to ship this on zod, then we can just poke %radar

Reply:
Perhaps it should just [send a |hi](https://github.com/urbit/urbit/blob/master/pkg/arvo/lib/hood/helm.hoon#L131-L132)? Though then you might get users saying "help! ~x is spamming my ship".

Reply:
Yeah, this is the original reason for poking some other app.

Reply:
Then we should probably just ship this, turn it on by default, and have this poke itself? Depending on "some other app" seems very unclean. (Also gives people a more-clear way of opting out of this, by stopping/disarming the radar app itself.)

Reply:
Im actually not sure we should ship this on ~zod, but if we are it might make sense to change the directionality of these pings, so that the users ping someone (their star or galaxy?) to say they're alive and therefore have the ability to opt out.

Reply:
At that point, something tells me ames already has that info. (^:

But yes, agree about that directionality change in a "shipping this" context.

Reply:
Good point, so maybe the thing to do eventually is to expose an ames scry for this data, and then have a simple app for requesting/collating other peoples knowledge of the network status.

Not sure what the timeline for this is, but if the bridge people need it soon, then keeping this on some branch other than master, and having it poke some other app seems sufficient in the near term.

Reply:
Yeah, most of this PR seems very redundant with ames' own pinging, which will recur continuously after you `|hi` a ship once. (And indeeed does not have a DELETE.) See also: "not responding still trying" printfs.

I think there's even an old "network status" app that pulled first/last seen timestamps out of the relevant ames `.^`?

Reply:
Looks like there was [code in womb](https://github.com/urbit/urbit/commit/b24bd2fcbf8c6ef418bc73f8802ddcbb23cc2fb1#diff-daa77bb57d3c030d205787e2fa5f5c10L304-L310) that iirc was used to show responding/stale ships as green vs yellow in a network status view. We'd probably want this as a single scry instead of a loop over `.^((list @p) %a /=pals=)`, and for performance reasons ames ping frequency has gotten turned down some which might be useful to adjust via this sort of interface - but letting ames keep track of PKI status seems desirable architecturally.

Reply:
/cc @belisarius222

Reply:
A reason to keep the directionality as-is is that it lets anyone use the radar app to get their own view of who's up.  Otherwise the information only flows in one direction (up), which seems incorrect.  This gives us the flexibility to put this app wherever we want.

Opting out by turning off radar seems reasonable.